pkg/api/scylla/validation/scylladbcluster_validation.go

```diff
@@ -0,0 +1,450 @@
+			scyllav1alpha1.NodeServiceTypeHeadless,
+			scyllav1alpha1.NodeServiceTypeLoadBalancer,
+		},
+		scyllav1alpha1.BroadcastAddressTypeServiceLoadBalancerIngress: {
+			scyllav1alpha1.NodeServiceTypeLoadBalancer,
+		},
+	}
+
+	allErrs = append(allErrs,
+		ValidateScyllaDBDatacenterBroadcastOptions(
+			options.Clients.Type,
+			SupportedScyllaV1Alpha1ScyllaDBClusterBroadcastAddressTypes,
+			scyllav1alpha1.NodeServiceTypeHeadless,
+			nodeServiceType,
+			allowedNodeServiceTypesByBroadcastAddressType,
+			fldPath.Child("clients"),
+		)...,
+	)
+
+	allErrs = append(allErrs,
+		ValidateScyllaDBDatacenterBroadcastOptions(
+			options.Nodes.Type,
+			SupportedScyllaV1Alpha1ScyllaDBClusterBroadcastAddressTypes,
+			scyllav1alpha1.NodeServiceTypeHeadless,
+			nodeServiceType,
+			allowedNodeServiceTypesByBroadcastAddressType,
+			fldPath.Child("nodes"),
+		)...,
+	)
+
+	return allErrs
+}
+
+func ValidateScyllaDBClusterUpdate(new, old *scyllav1alpha1.ScyllaDBCluster) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, ValidateScyllaDBCluster(new)...)
+	allErrs = append(allErrs, ValidateScyllaDBClusterSpecUpdate(new, old, field.NewPath("spec"))...)
+
+	return allErrs
+}
+
+func ValidateScyllaDBClusterSpecUpdate(new, old *scyllav1alpha1.ScyllaDBCluster, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(new.Spec.ClusterName, old.Spec.ClusterName, fldPath.Child("clusterName"))...)
+
+	oldDatacenterNames := slices.ConvertSlice(old.Spec.Datacenters, func(dc scyllav1alpha1.ScyllaDBClusterDatacenter) string {
+		return dc.Name
+	})
+	newDatacenterNames := slices.ConvertSlice(new.Spec.Datacenters, func(dc scyllav1alpha1.ScyllaDBClusterDatacenter) string {
+		return dc.Name
+	})
+
+	removedDatacenterNames := sets.New(oldDatacenterNames...).Difference(sets.New(newDatacenterNames...)).UnsortedList()
+	sort.Strings(removedDatacenterNames)
+
+	isDatacenterStatusUpToDate := func(sc *scyllav1alpha1.ScyllaDBCluster, dcStatus scyllav1alpha1.ScyllaDBClusterDatacenterStatus) bool {
+		return sc.Status.ObservedGeneration != nil && *sc.Status.ObservedGeneration >= sc.Generation && dcStatus.Stale != nil && !*dcStatus.Stale
+	}
+
+	for _, removedDCName := range removedDatacenterNames {
+		for i, oldDC := range old.Spec.Datacenters {
+			if oldDC.Name != removedDCName {
+				continue
+			}
+
+			oldDCNodeCount := controllerhelpers.GetScyllaDBClusterDatacenterNodeCount(old, oldDC)
+
+			if oldDCNodeCount != 0 {
+				allErrs = append(allErrs, field.Forbidden(fldPath.Child("datacenters").Index(i), fmt.Sprintf("datacenter %q can't be removed because it still has nodes that have to be scaled down to zero first", removedDCName)))
+				continue
+			}
+
+			oldDCStatus, _, ok := slices.Find(old.Status.Datacenters, func(dcStatus scyllav1alpha1.ScyllaDBClusterDatacenterStatus) bool {
+				return dcStatus.Name == removedDCName
+			})
+			if !ok {
+				continue
```

Comment:
shouldn't this also result in an error? Isn't no status equivalent to out-of-date status?